### PR TITLE
Update VisionEncoderDecoder to use an image processor

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Loading TrOCR processor failed because when checking the type of the feature extractor loaded, it was an image processor, rather than a feature extractor. 

This PR replaces: 
* Replaces the feature extractor with an image processor in `TrOCRProcessor` 
* Adds `AutoImageProcessor` to `AUTO_TO_BASE_CLASS_MAPPING` for the `ProcessorMixin` checks
* Adds backwards compatibility in case `feature_extractor` passed in as a kwarg when creating the processor. 
* Makes equivalent changes in `VisionEncoderDecoder` model

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/vision_text_dual_encoder/processing_vision_text_dual_encoder.py [hunk 1]
```diff
@@ -22,33 +22,43 @@
...
 
     Args:
-        feature_extractor ([`AutoFeatureExtractor`]):
-            The feature extractor is a required input.
+        image_processor ([`AutoImageProcessor`]):
+            The image processor is a required input.
         tokenizer ([`PreTrainedTokenizer`]):
             The tokenizer is a required input.
     """
-    feature_extractor_class = "AutoFeatureExtractor"
```
Comment (sgugger) [inline src/transformers/models/vision_text_dual_encoder/processing_vision_text_dual_encoder.py:38]: Same here.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this. Left some comments for the changes in the processors (there are also more processors to treat the same way, but can be done in a separate PR).

**Comment by ydshieh** [comment]
Thanks @amyeroberts 

Nothing extra (other than what Sylvain mentioned) from my side.

One thing I think would be great if you can provide a link to the line for
> when checking the type of the feature extractor loaded

Like:

See this line: 
https://github.com/huggingface/transformers/blob/c4cad8e3018e26f697f4ab0c5926e0c93aa0315b/src/transformers/processing_utils.py#L84

(For me, this way is easier to know what issue we have, and to see if the fix is good for the issue)

(Probably not necessary for Sylvain, as he knows everything in mind)

**Review by sgugger — APPROVED** [review]

File: src/transformers/models/trocr/processing_trocr.py [hunk 2]
```diff
@@ -25,28 +25,43 @@ class TrOCRProcessor(ProcessorMixin):
...
 
     Args:
-        feature_extractor ([`ViTFeatureExtractor`/`DeiTFeatureExtractor`]):
-            An instance of [`ViTFeatureExtractor`/`DeiTFeatureExtractor`]. The feature extractor is a required input.
+        image_processor ([`ViTImageProcessor`/`DeiTImageProcessor`]):
+            An instance of [`ViTImageProcessor`/`DeiTImageProcessor`]. The image processor is a required input.
         tokenizer ([`RobertaTokenizer`/`XLMRobertaTokenizer`]):
             An instance of [`RobertaTokenizer`/`XLMRobertaTokenizer`]. The tokenizer is a required input.
     """
-    feature_extractor_class = "AutoFeatureExtractor"
```
[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/models/trocr/processing_trocr.py:38]: Good point. I've added a property here: https://github.com/amyeroberts/transformers/blob/c28470b8ae3d3da700fae14bdfc3c53c07f1ffab/src/transformers/models/trocr/processing_trocr.py#L125

**Comment by amyeroberts** [comment]
@ydshieh That's a good point - thanks for the feedback! I'll make sure to add a link next time.

**Pull request merged**